### PR TITLE
feat(dependency_checker): add dependency checker script

### PR DESCRIPTION
**Pull request by a-maumau**
## Description
This PR will add a dependency checker script.
The script is originate in https://github.com/autowarefoundation/autoware/issues/3468 from @VRichardJP .
Some modification has done to address some packages.

The script will check whether some dependency in `package.xml` is used or not.

#### run output
```Text
~/autoware (main)$ ros2 run autoware_dependency_checker dependency_checker.sh 
--- Checking /home/masakibaba/autoware/src/universe/autoware.universe/evaluator/autoware_control_evaluator/package.xml ---
pluginlib seems not to be used

--- Checking /home/masakibaba/autoware/src/universe/autoware.universe/evaluator/autoware_planning_evaluator/package.xml ---
eigen should be rather be marked as <buildtool_depend>, not <depend>
pluginlib seems not to be used
tf2 seems not to be used

--- Checking /home/masakibaba/autoware/src/universe/autoware.universe/evaluator/autoware_evaluator_utils/package.xml ---
rclcpp_components seems not to be used

--- Checking /home/masakibaba/autoware/src/universe/autoware.universe/evaluator/diagnostic_converter/package.xml ---
pluginlib seems not to be used

--- Checking /home/masakibaba/autoware/src/universe/autoware.universe/evaluator/kinematic_evaluator/package.xml ---
autoware_planning_msgs seems not to be used
autoware_universe_utils seems not to be used
eigen should be rather be marked as <buildtool_depend>, not <depend>
tf2 seems not to be used

--- Checking /home/masakibaba/autoware/src/universe/autoware.universe/evaluator/perception_online_evaluator/package.xml ---
autoware_lanelet2_extension seems not to be used
autoware_vehicle_info_utils seems not to be used
eigen should be rather be marked as <buildtool_depend>, not <depend>
pluginlib seems not to be used
tf2 seems not to be used

--- Checking /home/masakibaba/autoware/src/universe/autoware.universe/evaluator/localization_evaluator/package.xml ---
autoware_planning_msgs seems not to be used
autoware_universe_utils seems not to be used
eigen should be rather be marked as <buildtool_depend>, not <depend>
tf2 seems not to be used

--- Checking /home/masakibaba/autoware/src/universe/autoware.universe/vehicle/autoware_external_cmd_converter/package.xml ---
autoware_raw_vehicle_cmd_converter seems not to be used
geometry_msgs seems not to be used
std_msgs seems not to be used
tier4_system_msgs seems not to be used

--- Checking /home/masakibaba/autoware/src/universe/autoware.universe/vehicle/autoware_accel_brake_map_calibrator/package.xml ---
autoware_motion_utils seems not to be used
autoware_raw_vehicle_cmd_converter seems not to be used
std_srvs seems not to be used
tf2_ros seems not to be used
...
```

It may not cover all the patterns in current autoware's including style.
Also, I wasn't sure where to place this script, so for now, it is located in the root directory.

## Related links
https://github.com/autowarefoundation/autoware/issues/3468

## Notes for reviewers
I am not sure with the maintainer, currently I put @youtalk . If there is a more appropriate person for a maintainer, let me know.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_dependency_checker/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+```Text
+# build
+$ cd to/autoware_tools
+$ colcon build --symlink-install --cmake-args --packages-up-to autoware_dependency_checker
+$ source
+
+# run
+$ cd to/your/autoware
+$ ros2 run ros2 run autoware_dependency_checker dependency_checker.sh
```
Comment (youtalk) [inline autoware_dependency_checker/README.md:29]:
```suggestion
$ ros2 run autoware_dependency_checker dependency_checker.sh
```

File: autoware_dependency_checker/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+# run in some package
+$ cd to/some/package
+$ ros2 run ros2 run autoware_dependency_checker dependency_checker.sh
```
Comment (youtalk) [inline autoware_dependency_checker/README.md:33]:
```suggestion
$ ros2 run autoware_dependency_checker dependency_checker.sh
```

File: autoware_dependency_checker/package.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>autoware_dependency_checker</name>
+  <version>0.1.0</version>
+  <description>The autoware_dependency_checker package</description>
+  <maintainer email="yutaka.kondo@youtalk.jp">Yutaka Kondo</maintainer>
+  <license>Apache License 2.0</license>
+  <author email="richard-v@macnica.co.jp">Vincent Richard</author>
+  <author email="masaki.baba@tier4.jp">Masaki Baba</author>
+
```
Comment (youtalk) [inline autoware_dependency_checker/package.xml:11]: Please insert `ament_cmake` dependency here.

**Review by youtalk — COMMENTED** [review]
Thank you for creating this PR!

**Comment by youtalk** [comment]
@a-maumau Please modify the shell scripts according to the https://results.pre-commit.ci/run/github/740824288/1722581617.pfcCvyu0QBWOhpHY9shZxQ.

File: autoware_dependency_checker/package.xml (hunk 3 above)
Comment (a-maumau) [reply autoware_dependency_checker/package.xml:11]: Thank you for the suggestion!
I fixed it at https://github.com/autowarefoundation/autoware_tools/pull/89/commits/45c7e3ea0b8b904d120a9981bd8f520c7d263e47

**Comment by a-maumau** [comment]
@youtalk I have fixed the sources based on pre-commit.

**Review by youtalk — APPROVED** [review]

**Comment by a-maumau** [comment]
sorry for bother you many times.
PR in the spell checker has merged (https://github.com/tier4/autoware-spell-check-dict/pull/757#issuecomment-2266515804), so it will pass the spell check.

**Comment by youtalk** [comment]
@a-maumau Graet work.  Thank you for the contribution.

**Pull request merged**